luci/luci-app-falter-owm/files/owm.sh

```diff
@@ -0,0 +1,318 @@
+	json_select $2
+	json_get_var localIP link_bindto
+	json_get_var remoteIP neighbor_originator
+	remotehost="$(nslookup $remoteIP | grep name | sed -e 's/.*name = \(.*\)/\1/' -e 's/\..*//')"".olsr"
+	# Maybe add some stuff here.
+	json_get_var linkQuality domain_metric_out_raw
+	#json_get_var linkQuality domain_metric_in_raw
+	json_get_var ifName "if"
+	json_select ..
+	olsr2links="$olsr2links$localIP $remoteIP $remotehost $linkQuality $ifName;"
+}
+
+olsr4_links() {
+	json_select $2
+	json_get_var localIP localIP
+	json_get_var remoteIP remoteIP
+	remotehost="$(nslookup $remoteIP | grep name | sed -e 's/.*name = \(.*\)/\1/')"
+	json_get_var linkQuality linkQuality
+	json_get_var olsrInterface olsrInterface
+	json_get_var ifName ifName
+	json_select ..
+	olsr4links="$olsr4links$localIP $remoteIP $remotehost $linkQuality $ifName;"
+}
+
+olsr6_links() {
+	json_select $2
+	json_get_var localIP localIP
+	json_get_var remoteIP remoteIP
+	remotehost="$(nslookup $remoteIP | grep name | sed -e 's/.*name = \(.*\)/\1/')"
+	json_get_var linkQuality linkQuality
+	json_get_var olsrInterface olsrInterface
+	json_get_var ifName ifName
+	json_select ..
+	olsr6links="$olsr6links$localIP $remoteIP $remotehost $linkQuality $ifName;"
+}
+
+# This section is relevant for hopglass statistics feature (isUplink/isHotspot)
+OLSRCONFIG=$(printf "/config" | nc localhost 9090)
+
+# collect nodes location
+uci_load system
+longitude="$(uci_get system @system[-1] longitude)"
+latitude="$(uci_get system @system[-1] latitude)"
+
+#
+#   Stop execution if lat/lon is not set.
+#
+if [ -z "$latitude" ] || [ -z "$longitude" ]; then
+	printf "latitude/longitude is not set.\nStopping now...\n"
+	exit 2
+fi
+
+
+# collect data on OLSR-links
+json_load "$(echo /nhdpinfo json link | nc ::1 2009 2>/dev/null)" 2>/dev/null
+olsr2links=""
+if json_is_a link array;then
+	json_for_each_item olsr2_links link
+fi
+json_cleanup
+json_load "$( echo /links | nc 127.0.0.1 9090 2>/dev/null)" 2>/dev/null
+#json_get_var timeSinceStartup timeSinceStartup
+olsr4links=""
+if json_is_a links array;then
+	json_for_each_item olsr4_links links
+fi
+json_cleanup
+json_load "$( echo /links | nc ::1 9090 2>/dev/null)" 2>/dev/null
+#json_get_var timeSinceStartup timeSinceStartup
+olsr6links=""
+if json_is_a links array;then
+	json_for_each_item olsr6_links links
+fi
+json_cleanup
+
+
+# collect board info
+json_load "$(ubus call system board)"
+json_get_var model model
+json_get_var hostname hostname
+json_get_var system system
+json_select release
+json_get_var revision revision
+json_get_var distribution distribution
+json_get_var version version
+json_select ..
+json_load "$(ubus call system info)"
+json_get_var uptime uptime
+json_get_values loads load
+
+# if file freifunk_release is available, override version and revision
+if [ -f /etc/freifunk_release ]; then
+	. /etc/freifunk_release
+	distribution="$FREIFUNK_DISTRIB_ID"
+	version="$FREIFUNK_RELEASE"
+	revision="$FREIFUNK_REVISION"
+fi
+
```

Comment:
Does it make sense to use /etc/openwrt_release if /etc/freifunk_release does no exist?

Reply:
I think so. That way, we get at least the OpenWrt-release basis. I consider this better than having empty fields.

Reply:
I see the changes you made with printf instead of echo.  But I still don't see souring in /etc/openwrt_release iif freifunk_release does not exist.

Reply:
Oh, that you mean. That's not necessary, as we get the values in question already by the ubus-call in line 120.